usr/sbin/p11sak/p11sak.c

```diff
@@ -3481,27 +3481,181 @@ static const struct p11sak_objtype *find_certtype(CK_KEY_TYPE ktype)
     return NULL;
 }
 
-static CK_RV get_key_infos(CK_OBJECT_HANDLE key, CK_OBJECT_CLASS *class,
-                           CK_KEY_TYPE *ktype, CK_ULONG *keysize,
-                           char** label, char** typestr,
-                           const struct p11sak_objtype **keytype)
+static CK_RV get_common_name_value(CK_OBJECT_HANDLE obj, char *label,
+                                   char **common_name_value)
+{
+    CK_ATTRIBUTE attr = { CKA_VALUE, NULL, 0 };
+    X509 *x509 = NULL;
+    const CK_BYTE *tmp_ptr;
+    char *subj = NULL, *cn_tmp, *cn_tmp2 = NULL;
+    CK_RV rc;
+
+    rc = get_attribute(obj, &attr);
+    if (rc != CKR_OK) {
+        warnx("Failed to retrieve attribute CKA_VALUE from object "
+              "\"%s\": 0x%lX: %s", label, rc, p11_get_ckr(rc));
+        return rc;
+    }
+
+    if (attr.ulValueLen == 0 || attr.ulValueLen == CK_UNAVAILABLE_INFORMATION) {
+        warnx("CKA_VALUE of object \"%s\" is unavailable: 0x%lX: %s",
```

Comment:
Do we really want to print an error message in this case ? This scrambles any list output......
Since you continue without an error, I would also not print an error message here.